Correct test_consistency_svm

fhazubski-Intel opened the pull request:
Do not use NULL svm pointers.
Add a check for clEnqueueSVMMigrateMem.

Signed-off-by: Filip Hazubski <filip.hazubski@intel.com>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/OpenCL-CTS?pullRequest=1071) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/OpenCL-CTS?pullRequest=1071) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/OpenCL-CTS?pullRequest=1071) it.</sub>

fhazubski-Intel commented:
Currently _test_consistency_svm_ use `NULL` svm pointers. For all of the functions: `clEnqueueSVMMemcpy`, `clEnqueueSVMMemFill`, `clEnqueueSVMMap`, `clEnqueueSVMUnmap` the specification states that the call returns `CL_INVALID_VALUE` if the svm pointers are `NULL`. For example here for `clEnqueueSVMMemcpy`: https://www.khronos.org/registry/OpenCL/specs/3.0-unified/html/OpenCL_API.html#clEnqueueSVMMemcpy
Because in the test the expectation is that `CL_INVALID_OPERATION` is returned, I have updated the code to not use `NULL` svm pointers. This will prevent any implementation of OpenCL from returning the unexpected `CL_INVALID_VALUE` value.

Additionally I have added a check for `clEnqueueSVMMigrateMem` function as it seemed to be missing. Again, I ensured that only `CL_INVALID_OPERATION` is allowed to be returned here by the specification.

gwawiork commented:
The issue  in test_api consistency_svm test case, visible on platforms which do no support SVM feature. 
LGTM !

bashbaug reviewed (commented):
Nice catch, LGTM.  Thanks for adding the missing check for `clEnqueueSVMMigrateMem` also.

Needs a non-Intel review before merging.
